homeassistant/components/sun/__init__.py

```diff
@@ -40,19 +40,19 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     return True
 
 
-async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+async def async_setup_entry(hass: HomeAssistant, entry: SunConfigEntry) -> bool:
     """Set up from a config entry."""
-    hass.data[DOMAIN] = Sun(hass)
+    entry.runtime_data = Sun(hass)
     await hass.config_entries.async_forward_entry_setups(entry, [Platform.SENSOR])
     return True
 
 
-async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+async def async_unload_entry(hass: HomeAssistant, entry: SunConfigEntry) -> bool:
     """Unload a config entry."""
     if unload_ok := await hass.config_entries.async_unload_platforms(
         entry, [Platform.SENSOR]
     ):
-        sun: Sun = hass.data.pop(DOMAIN)
+        sun = entry.runtime_data
         sun.remove_listeners()
```

Comment:
This could be moved to `async_setup_entry` with `entry.async_on_unload(sun.remove_listeners)`.